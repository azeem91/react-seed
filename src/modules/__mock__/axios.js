const CurrencyList = 'https://api.bitfinex.com/v2/tickers?symbols=ALL';

const data = [[1,2,3],[4,5,6]]
module.exports = {
  get: jest.fn((url) => {
    switch (url) {
      case CurrencyList:
        return Promise.resolve({
          data: data
        });
    }
  })
};