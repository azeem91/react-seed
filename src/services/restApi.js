
import axios from "axios";
import { AppConstants } from "../constants";

export const GET = async (url, params, baseUrl) => {
    const data = await makeCall('get', url, params, baseUrl)
    return data;
}

// TODO : Improve for error handeling and other method types 

export const POST = (baseUrl, url, param = {}, body = {}) => { }

export const PUT = (baseUrl, url, param = {}, body = {}) => { }


export const DELETE = (baseUrl, url, param = {}, body = {}) => { }

const errorHandeler = () => { }

const validation = () => { }

const parseUrlParams = (params) => {
    let queryString = Object.keys(params).map(key => key + '=' + params[key]).join('&');
    return `?${queryString}`
}

const makeCall = async (...data) => {
    
    const [method, url, params = null, baseUrl = AppConstants.api, body = null] = data;
    const fullUrl = `${baseUrl}${url}${params ? parseUrlParams(params) : ''}`;
    try {
        const responce = await axios.get(fullUrl)
        return responce;
    } catch (e) {
        console.error(e)
        return null;
    }


}

export default GET;