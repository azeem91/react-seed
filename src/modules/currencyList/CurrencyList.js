import React, { Component } from "react";
import { Link } from "react-router-dom";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";

// API MW keep code clean and deal all API calls form one place
import { GET } from '../../services/restApi';

export default class CurrencyList extends Component {
  state = {
    currencyList: []
  };

  async componentDidMount() {
    try {
      const res = await GET('tickers', { symbols: 'ALL' })
      this.setState({ currencyList: res.data })
    } catch (e) {
      console.error(e);
      this.setState({ currencyList: [] })
    }
  }

  formatCurrencyName = (currencySymbol) => {
    return currencySymbol
      .split("")
      .slice(1, currencySymbol.length - 1)
      .join("")
  }

  render() {
    return (
      <div className="CurrencyList">
        <header className="CurrencyList-header">
          <h1 className="CurrencyList-title">React Seed</h1>
        </header>
        <List>
          {this.state.currencyList.map(currency => {
            return (
              <Link key={currency[0]} to={`/currencyDetails/${currency[0]}`}>
                <ListItem>
                  <ListItemText
                    primary={this.formatCurrencyName(currency[0])}
                    secondary={"volume : " + currency[8]}
                  />
                </ListItem>
              </Link>
            );
          })}
        </List>
      </div>
    );
  }
}
