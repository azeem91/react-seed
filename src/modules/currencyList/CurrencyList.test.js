import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import CurrencyList from './CurrencyList'
import axios from 'axios';
import { GET } from '../../services/restApi';
import MockAdapter from 'axios-mock-adapter';

const flushPromises = () => new Promise(resolve => setImmediate(resolve));

it('CurrencyList snapshot test', async () => {
  const wrapper = shallow(<CurrencyList />);
  await flushPromises();
  wrapper.update();
  expect(wrapper).toMatchSnapshot();
});