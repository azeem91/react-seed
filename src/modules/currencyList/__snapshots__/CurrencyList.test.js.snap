// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyList snapshot test 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CurrencyList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="CurrencyList-header"
>
          <h1
                    className="CurrencyList-title"
          >
                    React Seed
          </h1>
</header>,
        <WithStyles(List)>
          
</WithStyles(List)>,
      ],
      "className": "CurrencyList",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1
            className="CurrencyList-title"
>
            React Seed
</h1>,
          "className": "CurrencyList-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "React Seed",
            "className": "CurrencyList-title",
          },
          "ref": null,
          "rendered": "React Seed",
          "type": "h1",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="CurrencyList-header"
>
            <h1
                        className="CurrencyList-title"
            >
                        React Seed
            </h1>
</header>,
          <WithStyles(List)>
            
</WithStyles(List)>,
        ],
        "className": "CurrencyList",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="CurrencyList-title"
>
              React Seed
</h1>,
            "className": "CurrencyList-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "React Seed",
              "className": "CurrencyList-title",
            },
            "ref": null,
            "rendered": "React Seed",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
